index.html

```diff
@@ -223,14 +228,28 @@
         The term <dfn>webapp server</dfn> refers to server-side components of a <a>webapp</a>.
       </p>
       <p>
-        The term <dfn>push message</dfn> refers to an indication to a <a>webapp</a> that there is
-        new information for it from the <a>webapp server</a>, all or part of which MAY be contained
-        in the <a>push message</a> itself.
+        A <dfn>push message</dfn> is an indication to a <a>webapp</a> that there is new information
+        for it from the <a>webapp server</a>, all or part of which MAY be contained in the <a>push
+        message</a> itself.
+      </p>
+      <p>
+        A <dfn>push subscription</dfn> is a message delivery context that is established between
+        the <a>user agent</a> and the <a>push service</a> on behalf of a <a>service worker
+        registration</a>, and it is associated with that <a>service worker registration</a>. A
+        <a>push subscription</a> MUST NOT persist longer than its associated <a>service worker
+        registration</a>, but it MAY be terminated earlier.
+      </p>
+      <p>
+        A <a>push subscription</a> has an associated <dfn>endpoint</dfn>. It MUST be the absolute
+        URL exposed by the <a>push server</a> where the <a>webapp server</a> can send <a title=
+        "push message">push messages</a>. It MAY be the same for all <a title=
+        "push subscription">push subscriptions</a> associated with a <a>push service</a>.
       </p>
       <p>
-        The term <dfn>push subscription</dfn> refers to each logical channel aimed at delivering
-        <a title="push message">push messages</a> from an <a>webapp server</a> to a <a>webapp</a>,
-        which is created by a distinct subscription of such <a>webapp</a> via this Push API.
+        A <a>push subscription</a> has an associated <dfn>subscription id</dfn>. It is used by the
+        <a>webapp server</a> to indicate the target of the <a title="push message">push
+        messages</a> that it submits to the <a>push server</a>. Each pair of <a>subscription id</a>
+        and <a>endpoint</a> is expected to be globally unique.
```

Comment:
I think this can be a MUST. If there are reasonable exceptions that indicate the need for a SHOULD, let's have a separate issue with a discussion for that.